app/forms/candidate_interface/continuous_applications/course_choice_validator.rb

```diff
@@ -0,0 +1,16 @@
+module CandidateInterface
+  module ContinuousApplications
+    class CourseChoiceValidator < ActiveModel::Validator
+      def validate(record)
+        return unless record.wizard.current_application
+        return if record.wizard.edit?
+
+        exists = record.wizard.current_application.application_choices.joins(:course_option).exists?(course_option: { course_id: record.course.id })
```

Comment:
Would this work?

```ruby
record.wizard.current_application.contains_course?(record.course)
```

Reply:
Looks like it does the same thing alright!

This doesn't look like the best implementation of this, does it?
```ruby
  def contains_course?(course)
    potential_course_option_ids = CourseOption.where(course_id: course.id).map(&:id)
    current_course_option_ids = application_choices.map(&:course_option_id)

    potential_course_option_ids.intersect?(current_course_option_ids)
  end
```

Reply:
Yes, could probably be improved.

Reply:
What about: `record.wizard.current_application.courses.exists?(record.course.id)`?